lib/config/fileconf.go

```diff
@@ -652,7 +652,7 @@ func checkAndSetDefaultsForGCPMatchers(matcherInput []GCPMatcher) error {
 type JoinParams struct {
 	TokenName string           `yaml:"token_name"`
 	Method    types.JoinMethod `yaml:"method"`
-	Azure     AzureJoinParams  `yaml:"azure"`
+	Azure     AzureJoinParams  `yaml:"azure,omitempty"`
```

Comment:
Out of curiosity, why is this in the join params and not in the token like it is for AWS joining?

Reply:
These params hold things that only the joining node needs to know (which in this case is just the client ID of the managed identity to use).